File: Excel/700-799/787/787 Challenge.py
import pandas as pd

path = "700-799/787/787 Right Answer Selection.xlsx"

input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows=13)
test = pd.read_excel(path, usecols="E:H", skiprows=1, nrows=3).rename(columns=lambda col: col.replace('.1', ''))

r1 = input[input['No'].astype(str).str.match(r'^\d+$') | (input['Correct'] == 'Y')].drop(columns=['Correct'])
part1 = r1[r1['No'].astype(str).str.match(r'^\d+$')].reset_index(drop=True)
part2 = r1[~r1['No'].astype(str).str.match(r'^\d+$')].reset_index(drop=True)
result = pd.concat([part1, part2], axis=1)
result.columns = test.columns
result['No'] = result['No'].astype(int)

print(result.equals(test))

File: Excel/700-799/787/787 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/700-799/787/787 Right Answer Selection.xlsx", range = "A2:C14")
test  = read_excel("Excel/700-799/787/787 Right Answer Selection.xlsx", range = "E2:H5")

r1 = input %>% filter(str_detect(No, "^[0-9]+$") | Correct == "Y") %>% select(-Correct)

result = bind_cols(
  filter(r1, str_detect(No, "^[0-9]+$")),
  filter(r1, !str_detect(No, "^[0-9]+$"))
) %>% set_names(colnames(test)) %>% mutate(No = as.integer(No))

all.equal(result, test)
